tests/unit/package_managers/test_gomod.py

```diff
@@ -1513,3 +1513,56 @@ def test_missing_gomod_file(file_tree, tmp_path):
 
     with pytest.raises(PackageRejected, match=path_error_string):
         fetch_gomod_source(request)
+
+
+@pytest.mark.parametrize(
+    "dep_replacements, pkg_config, raises_error",
+    (
+        ([], [{"path": "."}], False),
+        (
+            [{"name": "github.com/pkg/errors", "type": "gomod", "version": "v0.8.1"}],
+            [{"path": "."}],
+            False,
+        ),
+        (
+            [],
+            [{"path": "bar"}, {"path": "foo"}],
+            False,
+        ),
+        (
+            [{"name": "github.com/pkg/errors", "type": "gomod", "version": "v0.8.1"}],
+            [{"path": "."}, {"path": "foo"}],
+            True,
+        ),
+    ),
+)
+@mock.patch("cachi2.core.package_managers.gomod._find_missing_gomod_files")
+@mock.patch("cachi2.core.package_managers.gomod._resolve_gomod")
+@mock.patch("cachi2.core.package_managers.gomod.RequestOutput")
+@mock.patch("cachi2.core.models.input.Request")
```

Comment:
`RequestOutput` is mocked here to avoid validation errors. Since we're only testing the errors that are raised because of dependency replacements, I though it was ok to just mock this and ignore the actual contents of the outpu.

Now, for `Request`, I'd be very glad to use the fixture, but I need different packages for each scenario. Any suggestions on how to do that with pytest.fixture? Or should I just create a generator function?

Reply:
AFAIK pydantic models are mutable by default, it should be ok to do `gomod_request.packages = ...`

At least in the test code, in real code I would stick to the "immutable" mentality :smile: 

> RequestOutput is mocked here to avoid validation errors.

Ah, fair enough

Reply:
YOU CAN DO THAT? :exploding_head:

Reply:
so are we keeping `mock_request` here and using `gomod_request` only for the other test?

Reply:
Good catch, I changed the previous test to also use it.